src/query/cursor.c

```diff
@@ -1001,7 +1001,7 @@ cursor_buffer_last_page (CURSOR_ID * cursor_id_p, VPID * vpid_p)
 	  return ER_FAILED;
 	}
 
```

Comment:
A direct pointer of the memory allocation for CURSOR ID is stored in **buffer_area**. Memory allocation/release is executed in association with pointer variable **buffer_area**. I think the pointer **buffer** is just a pointer variable to hold some CURSOR id buffer_area and perform cursor operations for it. Therefore, memcpy and free for the pointer **buffer** are not necessary and should not be done. It seems you got a great point on the bug fix. By the way, the above if statement that checks whether **buffer** is NULL doesn't seem meaningful and could be removed (It seems that someone is concerned that memcpy could fail).

Reply:
Thank you for the review